lib/collectors.js

```diff
@@ -2,6 +2,46 @@
 
 module.exports = (api) => {
 
+  api.metasync.collect = (expected) => {
+
+    const isCount = typeof(expected) === 'number';
+    const isKeys = Array.isArray(expected);
+    if (!(isCount || isKeys)) throw new Error('Unexpected type');
+    const keys = expected;
+    if (isKeys) expected = expected.length;
+    let count = 0;
+    let timeout = 0;
+    let timer = null;
+    let done = null;
+    const data = {};
+
+    const collector = (key, value) => {
+      if (isCount || keys.includes(key)) count++;
+      if (value instanceof Error) return done(value, data);
```

Comment:
> We can change contract to `(key, error, value)` but it will not be compatible with errback.

I think I'd prefer either that or using different functions to signal for a successful or erroneous  data collection, as in:

```js
const dc = metasync
  .collect(1)
  .done(handleDone);

fs.readFile(filename, 'utf8', (error, data) => {
  if (error) {
    dc.fail('data', error);
  } else {
    dc.pick('data', data);
  }
});
```

The former option does make using errbacks somewhat easier, though:

```js
const dc = metasync
  .collect(1)
  .done(handleDone);

fs.readFile(filename, 'utf8', dc.bind(null, 'data'));
```

but the downside is that it'll make it a bit awkward to use *without* async operations.

I'd appreciate to hear some input from @metarhia/angeli on this.